File: SDKs/googleapis.com/cloudtrace/v2/python/src/sdk/models/shared/module.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class Module:
    r"""Module
    Binary module.
    """
    
    build_id: Optional[TruncatableString] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('buildId') }})
    module: Optional[TruncatableString] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('module') }})
    

File: SDKs/googleapis.com/cloudtrace/v2/typescript/src/sdk/models/shared/module.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { TruncatableString } from "./truncatablestring";



// Module
/** 
 * Binary module.
**/
export class Module extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=buildId" })
  buildId?: TruncatableString;

  @SpeakeasyMetadata({ data: "json, name=module" })
  module?: TruncatableString;
}
